packages/react-urql/src/hooks/useQuery.ts

```diff
@@ -1,9 +1,8 @@
 /* eslint-disable react-hooks/exhaustive-deps */
-
 import { DocumentNode } from 'graphql';
-import { Source, pipe, subscribe, onEnd, onPush, takeWhile } from 'wonka';
-import { useState, useEffect, useCallback, useMemo } from 'react';
-
+import { Source, pipe, subscribe, onPush, takeWhile } from 'wonka';
+import { useCallback, useMemo, useState } from 'react';
+import { useSyncExternalStore } from 'use-sync-external-store/shim';
```

Comment:
We may have to do some build tests on some bundlers to _ehem_ double check their work here